### ISSUE TITLE
[Bug] The definition of the center of the trust region is wrong

### ISSUE THREAD

**Issue by Luplz** [pytorch/botorch#2032]
# 🐛 Bug

In [scalable_constrained_bo.ipynb](https://github.com/pytorch/botorch/blob/main/tutorials/scalable_constrained_bo.ipynb), **the center of the trust region** is wrong. It should be the best point (taking constraints into account) and not the maximum.

### Best point

Replace
```python
 # if the minimum voilation candidate is smaller than the violation of the incumbent 
 if min_violation < self.best_constraint_values.sum()  :
```
in the function `update_state` by
```python
 # if the minimum voilation candidate is smaller than the violation of the incumbent and the best constraint values is not valid (otherwise none of new candidates are not better)
 if min_violation < self.best_constraint_values.sum() and (not torch.all(self.best_constraint_values<=0)) :
```

### Center of the trust region 
Add 
```python
    best_x : Tensor = torch.ones(dim, ) * torch.inf
```
in the class`ScboState` and replace
```python
   x_center = X[Y.argmax(), :].clone()
```
in the function `generate_batch` by 
```python
   x_center = state.best_x.clone()
```

**Comment by dme65** [comment]
Thanks for reporting! I will put up a fix for this as well as https://github.com/pytorch/botorch/issues/2031.

**Comment by dme65** [comment]
Fixed in #2034.
